modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -332,6 +333,13 @@ public void start(@Nullable String cfg) {
             // Deploy all registered watches because all components are ready and have registered their listeners.
             metaStorageMgr.deployWatches();
 
+            baselineMgr.listenBaselineChange(ctx -> {
```

Comment:
This registration of the given listener looks weird to me. It seems that this can be done in `BaselineManager.start()`. What do you think?

Reply:
I moved it here because this listener must be registered after `notifyConfigurationListeners();` so we could react to baseline changes only after proper start. In general, this listener can be registered somewhere in the init phase, development of which is still in progress.

Reply:
Please add a comment here in order to be sure that this code will improved in the future.